src/main/java/com/checkmarx/flow/service/MergeResultEvaluatorImpl.java

```diff
@@ -119,4 +114,49 @@ private static boolean exceedsThreshold(Map.Entry<FindingSeverity, Integer> find
         result.put(FindingSeverity.LOW, 0);
         return result;
     }
+
+    private static void writeMapToLog(Map<FindingSeverity, Integer> map, String message) {
+        String json;
+        try {
+            json = jsonMapper.writeValueAsString(map);
+        } catch (JsonProcessingException e) {
+            json = "<error>";
+            log.error("JSON serialization error.", e);
+        }
+        log.info("{}: {}", message, json);
+    }
+
+    private static void logThresholdCheck(
+            boolean exceedsThreshold, FindingSeverity severity, Integer threshold, int findingCount) {
+
+        if (threshold == null) {
+            log.debug("Threshold for the {} severity is not defined, skipping.", severity.name());
+        } else {
+            String message;
+            if (exceedsThreshold) {
+                message = "Finding count ({}) is above the threshold ({}) for the {} severity.";
+            } else {
+                message = "Finding count ({}) does not exceed the threshold ({}) for the {} severity.";
+            }
+            log.debug(message, findingCount, threshold, severity.name());
+        }
```

Comment:
Won't it be valuable to have an indication also in INFO level? (Aviv quite wanted it..)

Reply:
Done.